memdb.go

```diff
@@ -52,6 +52,10 @@ func NewMemDB(schema *DBSchema) (*MemDB, error) {
 	return db, nil
 }
 
+func (db *MemDB) DBSchema() *DBSchema {
+	return db.schema
```

Comment:
Is there a chance that someone might accidentally misuse this and modify the schema, affecting behavior during runtime?

Reply:
That is a good point.

Do you reckon it would be better to make and return a copy of the schema. The only downside would be whether one expects method called `DBSchema()` to do copying but I can't think of a good/better name here.

Reply:
Also I guess I can add inline docs to be more upfront about what the method does, but of course not everyone will read it.

Reply:
If feels idiomatic to just note that this should not be modified. I don't think we have a safe way to deep copy this (because `Indexers` ultimately can be anything), and I typically find that "deep copying" is left up to the caller anyways in most Go APIs.

Reply:
Good call - I added a comment!